### DISCUSSION TITLE
How to build testcontainers-java on ARM Mac?

### DISCUSSION THREAD

**Discussion by dawngerpony** [testcontainers/testcontainers-java#7544] [Q&A] [answered]
Based on my comment [here](https://github.com/testcontainers/testcontainers-java/issues/7448#issuecomment-1732500049), I am unable to run `./gradlew build` or `./gradlew check` on an M2 Mac without a failure from `azure-cosmos-emulator`. What's the best way for me to build testcontainers-java on an ARM Mac? I'm keen to contribute!

**Comment by eddumelendez** [comment, marked as answer]
Hi, don't need to build the whole project. I wonder what's the reason. If interested in contributing, first, please discuss your idea with the team. Do yo want to try a specific module? Try `./gradlew :<module>:check`. It could still be possible that some modules use image versions without arm support, in that case, check docker hub instead of reporting issues against those images.

**Reply by dawngerpony** [reply]
I was only trying to build the entire project locally because I thought it would be good hygiene to get  full green build before making a contribution. After working through additional similar issues with couchbase, cassandra, db2 etc I am starting to see that this project is not going to build fully without quite a lot of work, and potentially ignoring a bunch of tests on ARM architecture Macs. So I might give up and just raise my PR! Thanks 🙏
